qrl_testing/IntegrationTest.py

```diff
@@ -25,20 +25,75 @@
     "cp: cannot stat '/home/travis/genesis.yml': No such file or directory"
 }
 
-LogEntry = namedtuple('LogEntry', 'full node_id time version sync_state rest')
+LogEntry = namedtuple('LogEntry', 'full node_id time version synced rest')
+
+class NodeState:
+    def __init__(self, node_id):
+        self.node_id = node_id
+        self.ip = ''
+        self.Qaddress = ''
+        self.synced = False
+        self.grpc_started = False
+
+    def __repr__(self):
+        return "<NodeState ip: {} Qaddress: {} synced: {} grpc_started: {}>".format(self.ip, self.Qaddress, self.synced, self.grpc_started)
+
+    def find_ip_Qaddress_wallet(self):
+        tests_integration_path = os.path.dirname(os.path.dirname(__file__))
+        volumes_path = os.path.join(tests_integration_path, 'volumes/', 'testsintegration_{}'.format(self.node_id))
+
+        ip_file = os.path.join(volumes_path, "node_ip")
+        with open(ip_file) as f:
+            self.ip = f.readline().strip()
+
+        wallet_address_file = os.path.join(volumes_path, "wallet_address")
+        with open(wallet_address_file) as f:
+            self.Qaddress = f.readline().strip()
+        
+        self.wallet_dir = os.path.join(volumes_path, "wallet/")
+
+    def update(self, log_entry: LogEntry):
+        if log_entry.synced == 'synced':
+            self.synced = True
```

Comment:
This whole change implies that the only state that is important is synced... But this is a base class for all tests and we do care about all other states. The idea of cleaning up and avoiding strings is good but let's make them constants.. Let's not hide the fact that we have many possibilities.